exec-engine.md

```diff
@@ -0,0 +1,146 @@
+There may be subtle tweaks, beta starts in a few weeks*
+
+### `engine_forkchoiceUpdatedV1`
+
+This updates which L2 blocks the engine considers to be canonical (`forkchoiceState` argument),
+and optionally initiates block production (`payloadAttributes` argument).
+
+Within the rollup, the types of forkchoice updates translate as:
+- `headBlockHash`: block-hash of the head of the canonical chain. Labeled `"unsafe"` in user JSON-RPC.
+   Nodes may apply L2 blocks out of band ahead of time, and then reorg when L1 data conflicts.
+- `safeBlockHash`: block-hash of the canonical chain, derived from L1 data, unlikely to reorg.
+- `finalizedBlockHash`: irreversible block-hash, matches lower boundary of the dispute-period.
+
+To support rollup-functionality, one backwards-compatible change is introduced
+to [`engine_forkchoiceUpdatedV1`][engine_forkchoiceUpdatedV1]:
+
+[`PayloadAttributesV1`][PayloadAttributesV1] is extended with a `transactions` field,
+equivalent to the `transactions` field in [`ExecutionPayloadV1`][ExecutionPayloadV1]:
+> `Array of DATA` - Array of transaction objects, each object is a byte list (`DATA`) representing
+> `TransactionType || TransactionPayload` or `LegacyTransaction` as defined in [EIP-2718][eip-2718].
+
+This `transactions` field is an optional JSON field:
+- If empty or missing: no changes to engine behavior.
+  Utilized by sequencers (if enabled) to consume the transaction-pool.
+- If present and non-empty: the payload MUST only be produced with this exact list of transactions.
+  Utilized by block-derivation to compute full block payloads based on deterministic inputs.
+
+
+### `engine_executePayloadV1`
+
+No modifications to [`engine_executePayloadV1`][engine_executePayloadV1].
+Applies a L2 block to the engine state.
+
+### `engine_getPayloadV1`
+
+No modifications to [`engine_getPayloadV1`][engine_getPayloadV1].
+Retrieves a payload by ID, prepared by `engine_forkchoiceUpdatedV1` when called with `payloadAttributes`.
+
+## Networking
+
+The execution-engine can acquire all data through the rollup-node, as derived from L1:
+*P2P networking is strictly optional.*
+
+However, to provide L1-equivalent experience, the P2P network functionality SHOULD be enabled, serving:
+- Peer discovery ([Disc v5][discv5])
+- [`eth/66`][eth66]:
+  - Transaction pool (consumed by sequencer nodes)
+  - State-sync (happy-path for fast trustless db replication)
+  - Historical block header and body retrieval
+  - *New blocks are acquired through the consensus-layer instead (rollup-node)*
+
+No modifications to L1 network functionality are required, except configuration:
+- `networkID`: Distinguishes the L2 network from L1 and testnets. 
+  Equal to the `chainID` of the rollup network.
+- Activate Merge fork: Enables Engine API and disables propagation of blocks, 
+  as block-headers cannot be authenticated without consensus-layer.
+- Bootnode list: although DiscV5 is a shared network, bootstrap is faster through L2 nodes. 
```

Comment:
Brevity is valuable too. Especially for things that are widely implemented and specified elsewhere. I'll try to find some links for the curious reader instead.

Reply:
I think we must keep in mind that the target audience includes people that are not hardcore into implementation of Ethereum.

Especially since we really encourage reusing existing components, we must assume users will not necessarily have read the specs for those. Even if they have, some context is valuable to get situated.

I'm personally confused (but have not read the networking specs at this point). I don't consider myself to be particularly brilliant, but I think it would be very dangerous to assume that our "target reader" will understand things better than I do at this stage.

I think generally we should err of the side of over-explaining rather than under-explaining. Or said otherwise, lower the expectations we have of our readers.